packages/app/src/specs/generators/ExpandableFileChooser.vue

```diff
@@ -0,0 +1,166 @@
+<!--
+/**==============================================
+ * *          ExpandableFileChooser.vue
+ * Filter a list of files by a mix of glob pattern
+ * and a search string that includes a file's relative
+ * path.
+ *
+ * When clicked, the files will expand to display the
+ * #expanded-content slot.
+ *
+ * Features to note: debouncing + loading + row expanding
+ * ==============================================
+ * *                Debouncing
+ *
+ * ? To prevent calling update too frequently we debounce
+ * ? both the "No Results" UI and the update event
+ *
+ * ============================================**/
+-->
+
+<template>
+  <CreateSpecModalBody
+    variant="bare"
+    class="bg-white flex flex-col px-24px
+  relative"
+  >
+    <FileMatch
+      ref="fileMatchRef"
+      v-model:pattern="filePathSearch"
+      v-model:extensionPattern="localExtensionPattern"
+      class="bg-white pt-24px pb-12px top-0px z-10 sticky"
+      :matches="matches"
+    >
+      <template
+        v-if="loading"
+        #matches
+      >
+        <i-cy-loading_x16 class="h-24px mr-10px animate-spin w-24px" />
+      </template>
+    </FileMatch>
+
+    <div
+      v-show="loading"
+      data-testid="loading"
+    >
+      <!-- TODO(ryan) UNIFY-865: Get mocks for a loading state here -->
+      Loading
+    </div>
+    <ExpandableFileList
+      v-show="!loading"
+      :style="{ paddingTop: `${fileMatchHeight + 36}px` }"
+      class="right-24px left-24px absolute"
+      :files="filteredFiles"
+      :search="filePathSearch"
+    >
+      <template #expanded-content="{file}">
+        <ReactComponentList
+          :file="file"
+          @selectItem="selectItem"
+        />
+      </template>
+      <template #no-results>
+        <NoResults
+          empty-search
+          :search-term="noResults.search"
+          :message="noResults.message"
+          @clear="noResults.clear"
+        />
+      </template>
+    </ExpandableFileList>
+  </CreateSpecModalBody>
+</template>
+
+<script setup lang="ts">
+import { computed, ref } from 'vue'
+import { debouncedWatch, useDebounce, useElementSize } from '@vueuse/core'
+import { useI18n } from '@cy/i18n'
+
+import NoResults from '@cy/components/NoResults.vue'
+import CreateSpecModalBody from './CreateSpecModalBody.vue'
+import ExpandableFileList from './ExpandableFileList.vue'
+import FileMatch from '../../components/FileMatch.vue'
+import ReactComponentList from './component/ReactComponentList.vue'
+import { gql } from '@urql/core'
+
+const props = withDefaults(defineProps<{
+  files: any[]
+  extensionPattern: string
+  loading?: boolean
+}>(), {
+
+  loading: false,
+})
+
+const { t } = useI18n()
+
+gql`
+fragment FileChooser on FileParts {
+  relative
+  id
+  ...FileListItem
+}
+`
+
+const emits = defineEmits<{
+  (eventName: 'selectItem', { file: File, item: string })
```

Comment:
I think this is using the global `File` type: https://developer.mozilla.org/en-US/docs/Web/API/File

I would highly recommend picking a type that doesn't collide with `File`, I have been burnt by this before :sweat_smile:

Reply:
Yeah the syntax here was incorrect anyway which is probably why I didn't notice 👍🏻 good catch